### PR TITLE
Confidential VM features in VM and Disk cmdlets

### PR DESCRIPTION
**Pull request by Sandido**
## Description

This PR adds the Confidential VM functionality to the cmdlets:

1. Set-AzVMOSDisk
2. New-AzDiskEncryptionSetConfig
3. Set-AzDiskSecurityProfile
4. Set-AzVmSecurityProfile
5. Set-AzVmssSecurityProfile
6. Set-AzVmssStorageProfile

The SecurityType parameter was either added or expanded to use the 'Trusted Launch' value, and other related functionality to use a Disk Encryption Set with a Trusted Launch VM. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1187
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1140
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1138
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1136
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1137
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1197
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by NoriZC** [comment]
Hi @Sandido, as this draft PR hasn't been taken care of for long, could you please check and close it? Thanks!

File: src/Compute/Compute/VirtualMachine/Config/SetAzureVMSecurityProfile.cs [hunk 1]
```diff
@@ -39,8 +39,9 @@ public class SetAzureVMSecurityProfile : Microsoft.Azure.Commands.ResourceManage
...
         [Parameter(
            Mandatory = false,
-           ValueFromPipelineByPropertyName = true)]
-        [PSArgumentCompleter("TrustedLaunch")]
+           ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Specifies the SecurityType of the virtual machine. It has to be set to any specified value to enable UefiSettings. By default, UefiSettings will not be enabled unless this property is set. Possible values are 'ConfidentialVM' and 'TrustedLaunch'.")]
+        [PSArgumentCompleter("TrustedLaunch", "ConfidentialVM")]
         public string SecurityType { get; set; }
 
         public override void ExecuteCmdlet()
```
Comment (Sandido) [inline src/Compute/Compute/VirtualMachine/Config/SetAzureVMSecurityProfile.cs:47]: below part of the cmdlet can be improved.

File: src/KeyVault/KeyVault/Commands/Key/AddAzureKeyVaultKey.cs [hunk 2]
```diff
@@ -335,6 +335,12 @@ public class AddAzureKeyVaultKey : KeyVaultCmdletBase
             ParameterSetName = HsmInputObjectCreateParameterSet)]
         [Parameter(Mandatory = false,
             ParameterSetName = HsmResourceIdCreateParameterSet)]
+        [Parameter(Mandatory = false,
```
Comment (Sandido) [inline src/KeyVault/KeyVault/Commands/Key/AddAzureKeyVaultKey.cs:338]: Needs a bit of cleanup.

File: src/Compute/Compute/help/Set-AzVMOSDisk.md [hunk 3]
```diff
@@ -28,6 +28,7 @@ Set-AzVMOSDisk [-VM] <PSVirtualMachine> [[-Name] <String>] [[-VhdUri] <String>]
  [[-SourceImageUri] <String>] [[-CreateOption] <String>] [-Windows] [-DiskSizeInGB <Int32>]
  [-ManagedDiskId <String>] [-StorageAccountType <String>] [-DiskEncryptionSetId <String>] [-WriteAccelerator]
  [-DiffDiskSetting <String>] [-DiffDiskPlacement <String>] [-DeleteOption <String>]
+ [-SecurityEncryptionType <String>] [-SecureVMDiskEncryptionSet <String>]
```
Comment (Sandido) [inline src/Compute/Compute/help/Set-AzVMOSDisk.md:31]: Need to add examples of the cmdlets for conf vms in all cmdlets used in tests.

File: src/KeyVault/KeyVault/help/Add-AzKeyVaultKey.md [hunk 4]
```diff
@@ -380,6 +383,44 @@ Tags           :
 
 Create a secure key in managed hsm named testmhsm. Its name is test-key and type is RSA. 
 
+### Example 10: Add a key for a Confidential VM to a key vault. 
+```powershell
+$keyVaultName = <Key Vault Name>
+$location = <Azure Region>
+$resourceGroupName = <Resource Group Name>
```
Comment (wyunchi-ms) [inline src/KeyVault/KeyVault/help/Add-AzKeyVaultKey.md:390]: There are some syntex check in example. Please make sure the example have the right syntex os PowerShell.

Comment (Sandido) [reply src/KeyVault/KeyVault/help/Add-AzKeyVaultKey.md:390]: looks like I was missing a ", should be fixed.

File: src/Compute/Compute/VirtualMachine/Config/SetAzureVMOSDiskCommand.cs [hunk 5]
```diff
@@ -192,6 +192,19 @@ public class SetAzureVMOSDiskCommand : ComputeClientBaseCmdlet
         [PSArgumentCompleter("Detach", "Delete")]
         public string DeleteOption { get; set; }
 
+        [Parameter(
+           Mandatory = false,
+           ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Sets the SecurityEncryptionType value on the managed disk of the VM. possible values include: TrustedLaunch, ConfidentialVM_DiskEncryptedWithCustomerKey, ConfidentialVM_VMGuestStateOnlyEncryptedWithPlatformKey, ConfidentialVM_DiskEncryptedWithPlatformKey")]
```
Comment (Sandido) [inline src/Compute/Compute/VirtualMachine/Config/SetAzureVMOSDiskCommand.cs:198]: P

File: src/Compute/Compute/help/Set-AzVmssStorageProfile.md [hunk 6]
```diff
@@ -21,7 +21,8 @@ Set-AzVmssStorageProfile [-VirtualMachineScaleSet] <PSVirtualMachineScaleSet>
  [[-Image] <String>] [[-VhdContainer] <String[]>] [-ImageReferenceId <String>] [-OsDiskWriteAccelerator]
  [-DiffDiskSetting <String>] [-DiffDiskPlacement <String>] [-ManagedDisk <String>]
  [-DiskEncryptionSetId <String>] [-DataDisk <VirtualMachineScaleSetDataDisk[]>]
- [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
Comment (Sandido) [inline src/Compute/Compute/help/Set-AzVmssStorageProfile.md:24]: remove>?

File: src/Compute/Compute/help/Set-AzDiskSecurityProfile.md [hunk 7]
```diff
@@ -34,6 +34,49 @@ New-AzDisk -ResourceGroupName 'ResourceGroup01' -DiskName 'Disk01' -Disk $diskco
...
+$desConfig = New-AzDiskEncryptionSetConfig -Location $Location -SourceVaultId $keyvaultId -KeyUrl $keyUrl -IdentityType SystemAssigned -EncryptionType $diskEncryptionType;
+New-AzDiskEncryptionSet -ResourceGroupName $ResourceGroupName -Name $DesName -DiskEncryptionSet $desConfig;
+$diskencset = Get-AzDiskEncryptionSet -ResourceGroupName $ResourceGroupName -Name $desName;
+
+# Assign DES Access Policy to key vault
+$desIdentity = (Get-AzDiskEncryptionSet -Name $DesName -ResourceGroupName $ResourceGroupName).Identity.PrincipalId;
+Set-AzKeyVaultAccessPolicy -VaultName $KeyVaultName -ResourceGroupName $ResourceGroupName -ObjectId $desIdentity -PermissionsToKeys wrapKey,unwrapKey,get -BypassObjectIdValidation;
+        
+$diskSecurityType = "ConfidentialVM_DiskEncryptedWithCustomerKey";
+$diskName = "diskname";
```
Comment (Sandido) [inline src/Compute/Compute/help/Set-AzDiskSecurityProfile.md:70]: make gen eric

**Pull request merged**